profile.js

```diff
@@ -69,97 +36,9 @@ if (cli.input.length === 0) {
 	throw new Error('Specify a test file');
 }
 
-const file = path.resolve(cli.input[0]);
-const cacheDir = conf.cacheEnabled === false ? uniqueTempDir() : path.join(projectDir, 'node_modules', '.cache', 'ava');
-
-const babelConfig = babelPipeline.validate(conf.babel);
-conf.extensions = normalizeExtensions(conf.extensions || [], babelConfig);
-
-const _regexpFullExtensions = new RegExp(`\\.(${conf.extensions.full.map(ext => escapeStringRegexp(ext)).join('|')})$`);
-
-const precompileFull = babelPipeline.build(process.cwd(), cacheDir, babelConfig, conf.compileEnhancements === true);
-
-let precompileEnhancementsOnly = () => null;
-if (conf.compileEnhancements) {
-	precompileEnhancementsOnly = babelPipeline.build(projectDir, cacheDir, null, conf.compileEnhancements);
-}
-
-const precompiled = {};
-precompiled[file] = _regexpFullExtensions.test(file) ? precompileFull(file) : precompileEnhancementsOnly(file);
-
-const opts = {
-	file,
-	projectDir,
-	failFast: cli.flags.failFast,
-	serial: cli.flags.serial,
-	tty: false,
-	cacheDir,
-	precompiled,
-	require: resolveModules(conf.require)
-};
-
-// Mock the behavior of a parent process
-process.connected = true;
-process.channel = {ref() {}, unref() {}};
-
-const reporter = new VerboseReporter({
-	reportStream: process.stdout,
-	stdStream: process.stderr,
-	watching: false
-});
-
-const runStatus = new RunStatus(1, null);
-runStatus.observeWorker({
-	file,
-	onStateChange(listener) {
-		const emit = evt => listener(Object.assign(evt, {testFile: file}));
-		process.send = data => {
-			if (data && data.ava) {
-				const evt = data.ava;
-				if (evt.type === 'ping') {
-					if (console.profileEnd) {
-						console.profileEnd();
-					}
-
-					if (process.exitCode) {
-						emit({type: 'worker-failed', nonZeroExitCode: process.exitCode});
-					} else {
-						emit({type: 'worker-finished', forcedExit: false});
-						process.exitCode = runStatus.suggestExitCode({matching: false});
-					}
-
-					setImmediate(() => {
-						reporter.endRun();
-						process.emit('message', {ava: {type: 'pong'}});
-					});
-				} else {
-					emit(data.ava);
-				}
-			}
-		};
-	}
-}, file);
-
-reporter.startRun({
-	failFastEnabled: false,
-	files: [file],
-	matching: false,
-	previousFailures: 0,
-	status: runStatus
-});
-
-process.on('beforeExit', () => {
-	process.exitCode = process.exitCode || runStatus.suggestExitCode({matching: false});
-});
-
-// The "subprocess" will read process.argv[2] for options
-process.argv[2] = JSON.stringify(opts);
-process.argv.length = 3;
-
-if (console.profile) {
-	console.profile('AVA test-worker process');
+// Prefer the local installation of AVA
+if (importLocal(__filename)) {
+	debug('Using local install of AVA');
+} else {
+	require('./lib/cli').run({fork: false});
```

Comment:
I think we can remove `profile.js` with this PR. It's pretty awkward and you could now run `npx ava --no-fork --inspect my-test.js` instead.

Then we can also remove the `overrideConfig` from the `lib/cli`.

Reply:
I was thinking it could be removed also. I removed it and updated the docs with the new command.